lib/TingOpenformatMethods.inc

```diff
@@ -23,7 +23,37 @@ class TingOpenformatMethods {
         $ret[] = $subject;
       }
     }
+    return $ret;
+  }
 
+  /* \brief parse searchCode,display elements
+   * @param array of stdObjects
+   * return array of form []['display','searchCode']
+   */
+  public static function _parseSearchCode($stdObjects) {
+    $ret = Array();
+    if (!is_array($stdObjects)) {
+      $stdObjects = array($stdObjects);
+    }
+    foreach ($stdObjects as $key => $object) {
+      if (isset($object->searchCode) || isset($object->value)) {
+        if (isset($object->searchCode) && isset($object->searchString)) {
+          if (!is_array($object->searchString)) {
+            $object->searchString = array($object->searchString);
+          }
+          foreach ($object->searchString as $searchString) {
+            $subject['searchCode'] = $object->searchCode->{'$'} . '="' . $searchString->{'$'} . '"';
```

Comment:
Du løber et array igennem men sætter kun det sidste ind i $subject['searchCode']?

Overvej at bruge end() i stedet for.

Reply:
hmm, ja det ser da underligt ud - må lige igennem det igen, for at lure, hvad meningen egentlig er...